src/Navigation/WPF/Navigation.WPF/Models/RegistrationInfo.cs

```diff
@@ -10,14 +10,61 @@ public class RegistrationInfo :
         public Type ViewType { get; set; }
 
 
-        public bool PreCreateInstance { get; set; } = false;
+        public bool PreCreateInstance { get; set; }
 
 
         /// <inheritdoc cref="INavigationRegistration.ResolveViewUsingDependencyInjection" />
-        public bool ResolveViewUsingDependencyInjection { get; set; } = false;
+        public bool ResolveViewUsingDependencyInjection { get; set; }
 
 
         //public bool OpenInNewWindow { get; set; } = false;
         //public bool OpenAsPopup { get; set; } = false;
+
+
+        private Func<bool> condition = () => true;
+        public Func<bool> Condition
+        {
+            get => condition;
+            set
+            {
+                if (value == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '??=' operator here.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)